gensim/models/word2vec.py

```diff
@@ -625,6 +657,53 @@ def load_word2vec_format(cls, fname, fvocab=None, binary=False, norm_only=True):
         result.init_sims(norm_only)
         return result
 
+    def intersect_word2vec_format(self, fname, binary=False):
+        """
+        Merge the input-hidden weight matrix from the original C word2vec-tool format
+        given, where it intersects with the current vocabulary. (No words are added to the
+        existing vocabulary, but intersecting words adopt the file's weights, and 
+        non-intersecting words are left alone.)
+
+        `binary` is a boolean indicating whether the data is in binary word2vec format.
+        """
+        counts = None
+        overlap_count = 0
+        logger.info("loading projection weights from %s" % (fname))
+        with utils.smart_open(fname) as fin:
+            header = utils.to_unicode(fin.readline())
+            vocab_size, vector_size = map(int, header.split())  # throws for invalid file format
+            if not vector_size == self.vector_size:
+                logger.error("incompatible vector sizes")
+                # TOCONSIDER: maybe truncation/smaller vectors still useful enough to merge?
+                return # TODO raise ValueError()?
```

Comment:
Yes, the TODO idea – raising a ValueError – is likely a better way for this method to fail. 

(There were also a few TODO comments predating any of my edits... so I may yet leave some as pointers to 'ripe spots for future work', rather as than indicators of must-fix/must-do gaps. Unless, of course, that's a disfavored practice on this project, or some other convention should take precedence.)